### PR TITLE
Support principals in source.principal.

### PR DESCRIPTION
**Pull request by pitlv2109**
This is for trust domain migration:
* Support principals in source.principal.
* Handle duplicated principals from * prefix.

### CODE REVIEW

**Comment by pitlv2109** [comment]
@liminw @yangminzhu ping.

File: pilot/pkg/security/authz/model/model.go [hunk 1]
```diff
@@ -157,7 +157,11 @@ func NewModelV1alpha1(trustDomainBundle trustdomain.Bundle, role *istio_rbac.Ser
 
 			property := KeyValues{}
 			for k, v := range subject.Properties {
-				property[k] = []string{v}
+				values := []string{v}
```
Comment (yangminzhu) [inline pilot/pkg/security/authz/model/model.go]: This shouldn't be done in `model.go`, can you move it to `principal.go`?

Comment (yangminzhu) [reply pilot/pkg/security/authz/model/model.go]: Oh, I didn't realize it's already added in other places, it's okay for this PR but could you put a TODO to refactor this?

Comment (yangminzhu) [reply pilot/pkg/security/authz/model/model.go]: To be more clear, the refactor is to add a `func (principal *Principal) ReplaceTrustDomainAliases() that searches and replace all trust domains in the principal. Then you just need to call it once in the end of `NewModelV1beta1()` or `NewModelV1alpha1()`.

Comment (pitlv2109) [reply pilot/pkg/security/authz/model/model.go]: Added a TODO. Feel free to remove the hold tag once all tests have passed. Thanks!

Comment (yangminzhu) [reply pilot/pkg/security/authz/model/model.go]: Thanks!

**Pull request merged**